tools/plugins/com.liferay.ide.service.core/src/com/liferay/ide/service/core/AddServiceBuilderOperation.java

```diff
@@ -53,27 +64,104 @@ public AddServiceBuilderOperation(IDataModel model) {
 	public void createDefaultServiceBuilderFile(IFile serviceBuilderFile, IProgressMonitor monitor)
 		throws CoreException {
 
-		String descriptorVersion = null;
+		String descriptorVersion = "7.0.0";
+
+		IProject project = serviceBuilderFile.getProject();
 
 		try {
-			ILiferayProject liferayProject = LiferayCore.create(ILiferayProject.class, serviceBuilderFile.getProject());
+			ILiferayProject liferayProject = LiferayCore.create(ILiferayProject.class, project);
 
 			ILiferayPortal portal = liferayProject.adapt(ILiferayPortal.class);
 
-			Version portalVersion = new Version(portal.getVersion());
+			Version portalVersion = null;
+
+			if (portal != null) {
+				portalVersion = new Version(portal.getVersion());
+			}
+			else {
+				IWorkspaceProject workspaceProject = LiferayWorkspaceUtil.getLiferayWorkspaceProject();
 
-			descriptorVersion = portalVersion.getMajor() + "." + portalVersion.getMinor() + ".0";
+				if (workspaceProject != null) {
+					String targetPlatformVersion = workspaceProject.getTargetPlatformVersion();
+
+					if (targetPlatformVersion != null) {
+						portalVersion = new Version(workspaceProject.getTargetPlatformVersion());
+					}
+				}
+			}
+
+			if (portalVersion != null) {
+				descriptorVersion = portalVersion.getMajor() + "." + portalVersion.getMinor() + ".0";
+			}
 		}
 		catch (Exception e) {
 			ProjectCore.logError("Could not determine liferay runtime version", e);
-			descriptorVersion = "6.0.0";
+			descriptorVersion = "7.0.0";
 		}
 
 		WizardUtil.createDefaultServiceBuilderFile(
 			serviceBuilderFile, descriptorVersion, getDataModel().getBooleanProperty(USE_SAMPLE_TEMPLATE),
 			getDataModel().getStringProperty(PACKAGE_PATH), getDataModel().getStringProperty(NAMESPACE),
 			getDataModel().getStringProperty(AUTHOR), monitor);
 
+		IBundleProject bundleProject = LiferayCore.create(IBundleProject.class, project);
+
+		if (bundleProject != null) {
+			WizardUtil.addBundleInfo(project);
+
+			IProjectBuilder projectBuilder = bundleProject.adapt(IProjectBuilder.class);
+
+			List<Artifact> artifacts = new ArrayList<>();
+
+			IWorkspaceProject workspaceProject = LiferayWorkspaceUtil.getLiferayWorkspaceProject();
+
+			boolean hasTargetPlatform = false;
+
+			if (workspaceProject != null) {
+				String targetPlatformVersion = workspaceProject.getTargetPlatformVersion();
+
+				if (targetPlatformVersion != null) {
+					hasTargetPlatform = true;
+				}
+			}
+
+			String bndlibVersion = "3.1.0";
```

Comment:
@jtydhr88  @Seiphon 

All of the following code is hard code (artifacts, versions, etc).  It will be very hard to keep this code updated.  We should come up with another method to update the build.gradle, such as, take a look at the content that is created by a new service builder project to see what versions are added to the -service project and then replicate those versions here.

Also we need a unit test to tests this behavior because we must protect against the future case where the service builder template might change significantly.